zcash_keys/src/keys.rs

```diff
@@ -71,6 +70,33 @@ pub mod sapling {
     }
 }
 
+/// A hash of a seed used for an HD account.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct HDSeedFingerprint([u8; BLAKE2B_LENGTH]);
+
+impl HDSeedFingerprint {
+    /// Generates the fingerprint from a given seed.
+    pub fn from_seed(seed: &SecretVec<u8>) -> Self {
```

Comment:
Instead of the inline panic I suggest in my previous comment, this should instead be changed to check the length requirements on seeds ("MUST be at least 32 and at most 252 bytes" for [Sapling](https://zips.z.cash/zip-0032#sapling-master-key-generation) and [Orchard](https://zips.z.cash/zip-0032#orchard-master-key-generation) master key generation), and either panic up-front or return an error.